internal/controller/ingress_controller.go

```diff
@@ -341,6 +371,62 @@ func (r *IngressReconciler) listIngressesBySecret(ctx context.Context, obj clien
 	return requests
 }
 
+func (r *IngressReconciler) listIngressForBackendTrafficPolicy(ctx context.Context, obj client.Object) (requests []reconcile.Request) {
+	v, ok := obj.(*v1alpha1.BackendTrafficPolicy)
+	if !ok {
+		r.Log.Error(fmt.Errorf("unexpected object type"), "failed to convert object to BackendTrafficPolicy")
+		return nil
+	}
+	var namespacedNameMap = make(map[types.NamespacedName]struct{})
+	ingresses := []networkingv1.Ingress{}
+	for _, ref := range v.Spec.TargetRefs {
+		service := &corev1.Service{}
+		if err := r.Get(ctx, client.ObjectKey{
+			Namespace: v.Namespace,
+			Name:      string(ref.Name),
+		}, service); err != nil {
+			if client.IgnoreNotFound(err) != nil {
+				r.Log.Error(err, "failed to get service", "namespace", v.Namespace, "name", ref.Name)
+			}
+			continue
+		}
+		ingressList := &networkingv1.IngressList{}
+		if err := r.List(ctx, ingressList, client.MatchingFields{
+			indexer.ServiceIndexRef: indexer.GenIndexKey(v.GetNamespace(), string(ref.Name)),
```

Comment:
Can I use the same index key for `HTTPRoute` and `Ingress`? I get an index key conflict error when using the same key

Reply:
The index of each resource is unique and cannot be created repeatedly, but it can be multiple